src/EventListener/MapToListener.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AutoMapper\EventListener;
+
+use AutoMapper\Attribute\MapFrom;
+use AutoMapper\Attribute\MapTo;
+use AutoMapper\Event\GenerateMapperEvent;
+use AutoMapper\Event\PropertyMetadataEvent;
+use AutoMapper\Event\SourcePropertyMetadata;
+use AutoMapper\Event\TargetPropertyMetadata;
+use AutoMapper\Exception\BadMapDefinitionException;
+use AutoMapper\Transformer\CallableTransformer;
+use AutoMapper\Transformer\CustomTransformer\CustomModelTransformer;
+use AutoMapper\Transformer\CustomTransformer\CustomModelTransformerInterface;
+use AutoMapper\Transformer\CustomTransformer\CustomPropertyTransformer;
+use AutoMapper\Transformer\CustomTransformer\CustomPropertyTransformerInterface;
+use AutoMapper\Transformer\CustomTransformer\CustomTransformersRegistry;
+use AutoMapper\Transformer\TransformerInterface;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+use Symfony\Component\String\Inflector\EnglishInflector;
+use Symfony\Component\String\Inflector\InflectorInterface;
+
+final readonly class MapToListener
+{
+    public function __construct(
+        private CustomTransformersRegistry $customTransformersRegistry,
+        private InflectorInterface $inflector = new EnglishInflector(),
+    ) {
+    }
+
+    public function __invoke(GenerateMapperEvent $event): void
+    {
+        if ($event->mapperMetadata->sourceReflectionClass === null) {
+            return;
+        }
+
+        $this->setPropertiesFromSource($event);
+        $this->setPropertiesFromTarget($event);
```

Comment:
shouldn't this `MapToListener` only handle `MapTo` attributes? and handle `MapFrom` in another listener?
Here, because of the previous condition, we cannot deal with "array to object" transformations (I don't know if you are planning to do this in another PR)